src/utils.rs

```diff
@@ -268,15 +268,37 @@ pub(crate) fn now_systime(clock: &dyn Clock) -> SystemTime {
         .expect("Failed to convert Clock time to SystemTime")
 }
 
+/// Computes the "index key" (lowest bound) for an SST index block, ie a key that's greater
+/// than all keys in the previous block and less than or equal to all keys in the new block
+pub(crate) fn index_key(prev_block_last_key: Option<Bytes>, this_block_first_key: &Bytes) -> Bytes {
```

Comment:
I think this fn needs to deal with the case where the previous block's last key is the same as the current block's first key - which can happen in the current impl for l0 ssts when there are updates. In that case we can just use the whole key as the index key.